### PR TITLE
fix(dev-server run): Check for environment tag not being undefined

### PR DESCRIPTION
**Pull request by sinhashubham95**
### SUMMARY
Environment tag is not set in the dev-server, so reference to a key in the environment tag was making the dev-server not loadable.

### TESTING INSTRUCTIONS
Try to load Superset on dev-server, it does not load. After this fix, it should load.

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/21230
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by sinhashubham95** [comment]
@villebro can you please check this?

**Comment by mtv2102** [comment]
@sinhashubham95 I tried as like your changes from your commit [https://github.com/apache/superset/pull/21232/commits/53f9f75d443bba7689a2a77473fed2e11b999e23]
but my issue still remains same...Is there any other place that I missed out...?

**Review by stephenLYZ — APPROVED** [review]
LGTM!

**Comment by sinhashubham95** [comment]
@villebro @stephenLYZ can you merge this?

File: superset-frontend/src/views/components/RightMenu.tsx [hunk 1]
```diff
@@ -281,7 +281,7 @@ const RightMenu = ({
           onDatabaseAdd={handleDatabaseAdd}
         />
       )}
-      {environmentTag.text && (
+      {environmentTag && environmentTag.text && (
```
Comment (villebro) [inline superset-frontend/src/views/components/RightMenu.tsx:284]: Could we simplify this to
```suggestion
      {environmentTag?.text && (
```

**Review by villebro — COMMENTED** [review]
Minor nit, other than that LGTM

File: superset-frontend/src/views/components/RightMenu.tsx (hunk 1 above)
Comment (villebro) [reply superset-frontend/src/views/components/RightMenu.tsx:284]: Oh, too late 🙁

Comment (stephenLYZ) [reply superset-frontend/src/views/components/RightMenu.tsx:284]: Ops...Let's put it on the next PR

Comment (villebro) [reply superset-frontend/src/views/components/RightMenu.tsx:284]: > Ops...Let's put it on the next PR

no worries, it was just a minor nit 🙂👍

**Pull request merged**